src/index.js

```diff
@@ -0,0 +1,238 @@
+    skyImage.alt =
+      'A dark sky with one big bolt of lightning stretching toward the ground';
+    skyCaption.textContent = 'Photo by Brandon Morgan';
+  } else if (selectedSky === 'snowy') {
+    skyImage.src = 'assets/sky/jessica-fadel-snowing.jpg';
+    skyImage.alt =
+      'A snow flurry against a dark background, as though it is nighttime';
+    skyCaption.textContent = 'Photo by Jessica Fadel';
+  }
+};
+
+const updateDisplayCity = () => {
+  const displayCity = document.getElementById('display-city');
+  const userInputCity = document.getElementById('input-city').value;
+
+  // update display city text content
+  displayCity.textContent = userInputCity;
+};
+
+// function to display real time temperature
+const showCurrentTemp = () => {
+  // update state current city based on completed user input
+  state.currentCity = document.getElementById('display-city').textContent;
+
+  const currentTemp = document.getElementById('temperature');
+
+  // update display temp, sky, and landscape after retrieving weather data
+  getCurrentTemp().then(() => {
+    currentTemp.textContent = `${state.temperature}° F`;
+
+    changeTemperatureColor();
+    changeLandscape();
+    changeSky();
+  });
+};
+
+// reset display city
+const resetDisplayCity = () => {
+  const displayCity = document.getElementById('display-city');
+
+  document.getElementById('input-city').value = '';
+  displayCity.textContent = 'Chicago';
+  state.currentCity = displayCity.textContent;
+
+  showCurrentTemp();
+};
+
+const registerEventHandlers = () => {
+  const increaseTempButton = document.getElementById('increase-temp');
+  increaseTempButton.addEventListener('click', increaseTemperature);
+
+  const decreaseTempButton = document.getElementById('decrease-temp');
+  decreaseTempButton.addEventListener('click', decreaseTemperature);
+
+  const userInputCity = document.getElementById('input-city');
+  userInputCity.addEventListener('input', updateDisplayCity);
+
+  const currentTempButton = document.getElementById('current-temp-button');
+  currentTempButton.addEventListener('click', showCurrentTemp);
+
+  const resetCityButton = document.getElementById('reset-city-button');
+  resetCityButton.addEventListener('click', resetDisplayCity);
+
+  const skyDropdown = document.getElementById('sky-dropdown');
+  skyDropdown.addEventListener('change', manuallyChangeSky);
+};
+
+document.addEventListener('DOMContentLoaded', registerEventHandlers);
+
+// endpoints to retrieve real time weather data
+
+const getCurrentTemp = () => {
+  // get latitude and longitude of current city
+  return (
+    axios
+      .get('http://127.0.0.1:5000/location', {
+        params: {
+          q: state.currentCity,
+        },
+      })
+      .then((response) => {
+        const lat = response.data[0]['lat'];
+        const lon = response.data[0]['lon'];
+
+        return [lat, lon];
+      })
+      // get current temp of current city
+      .then((response) => {
```

Comment:
Nice use of `.then` chaining to fire the calls off in sequence! This function is a bit long, if we wanted to divide it up for clarity, we could move the `/weather` endpoint call to its own function and call the new function from a `.then` block here.

Reply:
Ooh, I'd love to talk about this when we check in next!